core/src/test/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionMonitorSpec.scala

```diff
@@ -587,6 +587,15 @@ class SubmissionMonitorSpec(_system: ActorSystem) extends TestKit(_system) with
     }
   }
 
+  it should "stop trying to monitor a submission that's been deleted" in withDefaultTestDatabase { dataSource: SlickDataSource =>
+    val monitorRef = createSubmissionMonitorActor(dataSource, testData.submission1, testData.wsName, new SubmissionTestExecutionServiceDAO(WorkflowStatuses.Running.toString))
+    watch(monitorRef)
+
+    runAndWait(submissionQuery.delete(SlickWorkspaceContext(testData.workspace), testData.submission1.submissionId))
+
+    expectMsgClass(5 seconds, classOf[Terminated])
```

Comment:
Instead of just checking for termination, maybe we could check for the `SubmissionDeletedException` message itself. Might be a stronger test because even a healthy actor terminates itself within 5 seconds.

Maybe this would work:
`expectMsg(Failure(new SubmissionDeletedException))`
or
```
expectMsgPF(5 seconds) {
  case e@Failure(ex: SubmissionDeletedException) => e
}
```
http://doc.akka.io/docs/akka/2.5.3/scala/testing.html